src/VisualStudio/Core/Def/Implementation/Snippets/AbstractSnippetExpansionClient.cs

```diff
@@ -63,6 +107,13 @@ public AbstractSnippetExpansionClient(IThreadingContext threadingContext, Guid l
 
         public int FormatSpan(IVsTextLines pBuffer, VsTextSpan[] tsInSurfaceBuffer)
         {
+            // If this is a manually-constructed snippet for a full method call, avoid formatting the snippet since
+            // doing so will disrupt signature help.
+            if (!_state._methods.IsDefault)
```

Comment:
nit: consider a well named helper to provide this information.  like IsMethodCallArgumentExpansion (or whatever naming you've been using elsewhere).

Reply:
oh, there's already: IsFullMethodCallSnippet.  use that :)   consider making a helper property for that on this type as well.

Reply:
I'm fairly certain `FormatSpan` is called by the snippet engine before we have access to the expansion session. `IsFullMethodCallSnippet` would return false at this point.

Reply:
⏱ Updating documentaiton